source/integrations/zapier.rst

```diff
@@ -0,0 +1,147 @@
+- `Troubleshooting Guide <https://docs.mattermost.com/integrations/zapier.html#troubleshooting-guide>`_ - **Advice on troubleshooting common setup issues**
+
+
+Zapier Setup Guide
+--------------------------------
+Zapier is authorized using OAuth2.0. The setup guide requires that a System Admin register the Zapier app on their Mattermost server and can then optionally allow any users on the system to create Zapier integrations.
+
+Enable Zapier on your Mattermost Instance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The first time you set up Zapier on your Mattermost instance you will be required to enable an OAuth 2.0 application which can be accessible by everyone on your server. Your System Admin must execute these steps.
+
+Enable OAuth 2.0
+```````````````````````````
+1. Go to the **Main Menu** > **System Console**
+2. Under **Integrations** > **Custom Integrations**
+  a. Set `Enable OAuth 2.0 Service Provider <https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider>`_ to **True**.
+  b. If you’d like to allow Zapier integrations to post with customizable usernames and profile pictures, then set `Enable webhooks and slash commands to override usernames <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames>`_ and `Enable webhooks and slash commands to override profile picture icons <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-profile-picture-iconss>`_ to **True**.
+  c. If you’d like to allow users on your system who are not Team or System Admins to create Zapier integrations, then set `Restrict creating integrations to Team and System Admins <https://docs.mattermost.com/administration/config-settings.html#restrict-creating-integrations-to-team-and-system-admins>`_ to **False**.
+
+Register Zapier as an OAuth 2.0 Application
+````````````````````````````````````````````````````````````````
+1. Go to **Main Menu** > **Integrations**
+2. Click **OAuth 2.0 Applications**, then click **Add OAuth 2.0 Application** and enter the following fields:
+  a. **Is Trusted**: Yes
+  b. **Display Name**: ``Zapier``
+  c. **Description**: ``Application for Zapier integrations``
+  d. **Homepage**: ``https://zapier.com/``
+  e. **Icon URL**: ``http://bit.ly/2bxrzv0``
+  f. **Callback URLs**: ``https://zapier.com/dashboard/auth/oauth/return/App27274API/``
+3. Click **Save** to create the application. You will be provided with a **Client ID** and **Client Secret**. Save these values to connect Zapier in the steps below.
+
+.. image:: ../images/zapier-oauth.png
+
+
+Create a Zap
+~~~~~~~~~~~~~~~~~~~~~~~~~
+1. Click and accept this invitation to our Zapier app: https://zapier.com/developer/invite/27274/233f5629a5c215f542747a9c6d395b98/
+2. On your `Zapier dashboard <https://zapier.com/app/dashboard>`_ click **Make a Zap!**.
```

Comment:
At this point, I realised I needed a Zapier account.  Perhaps we need to clarify that?